bemani/backend/iidx/heroicverse.py

```diff
@@ -0,0 +1,1883 @@
+                self.GAME_SP_DAN_RANK_1_DAN: self.DAN_RANK_1_DAN,
+                self.GAME_SP_DAN_RANK_2_DAN: self.DAN_RANK_2_DAN,
+                self.GAME_SP_DAN_RANK_3_DAN: self.DAN_RANK_3_DAN,
+                self.GAME_SP_DAN_RANK_4_DAN: self.DAN_RANK_4_DAN,
+                self.GAME_SP_DAN_RANK_5_DAN: self.DAN_RANK_5_DAN,
+                self.GAME_SP_DAN_RANK_6_DAN: self.DAN_RANK_6_DAN,
+                self.GAME_SP_DAN_RANK_7_DAN: self.DAN_RANK_7_DAN,
+                self.GAME_SP_DAN_RANK_8_DAN: self.DAN_RANK_8_DAN,
+                self.GAME_SP_DAN_RANK_9_DAN: self.DAN_RANK_9_DAN,
+                self.GAME_SP_DAN_RANK_10_DAN: self.DAN_RANK_10_DAN,
+                self.GAME_SP_DAN_RANK_CHUDEN: self.DAN_RANK_CHUDEN,
+                self.GAME_SP_DAN_RANK_KAIDEN: self.DAN_RANK_KAIDEN,
+            }[game_dan]
+        elif cltype == self.GAME_CLTYPE_DOUBLE:
+            return {
+                self.GAME_DP_DAN_RANK_7_KYU: self.DAN_RANK_7_KYU,
+                self.GAME_DP_DAN_RANK_6_KYU: self.DAN_RANK_6_KYU,
+                self.GAME_DP_DAN_RANK_5_KYU: self.DAN_RANK_5_KYU,
+                self.GAME_DP_DAN_RANK_4_KYU: self.DAN_RANK_4_KYU,
+                self.GAME_DP_DAN_RANK_3_KYU: self.DAN_RANK_3_KYU,
+                self.GAME_DP_DAN_RANK_2_KYU: self.DAN_RANK_2_KYU,
+                self.GAME_DP_DAN_RANK_1_KYU: self.DAN_RANK_1_KYU,
+                self.GAME_DP_DAN_RANK_1_DAN: self.DAN_RANK_1_DAN,
+                self.GAME_DP_DAN_RANK_2_DAN: self.DAN_RANK_2_DAN,
+                self.GAME_DP_DAN_RANK_3_DAN: self.DAN_RANK_3_DAN,
+                self.GAME_DP_DAN_RANK_4_DAN: self.DAN_RANK_4_DAN,
+                self.GAME_DP_DAN_RANK_5_DAN: self.DAN_RANK_5_DAN,
+                self.GAME_DP_DAN_RANK_6_DAN: self.DAN_RANK_6_DAN,
+                self.GAME_DP_DAN_RANK_7_DAN: self.DAN_RANK_7_DAN,
+                self.GAME_DP_DAN_RANK_8_DAN: self.DAN_RANK_8_DAN,
+                self.GAME_DP_DAN_RANK_9_DAN: self.DAN_RANK_9_DAN,
+                self.GAME_DP_DAN_RANK_10_DAN: self.DAN_RANK_10_DAN,
+                self.GAME_DP_DAN_RANK_CHUDEN: self.DAN_RANK_CHUDEN,
+                self.GAME_DP_DAN_RANK_KAIDEN: self.DAN_RANK_KAIDEN,
+            }[game_dan]
+        else:
+            raise Exception('Invalid cltype!')
+
+    # Override base method from iidx 27 and on
+    def make_score_struct(self, scores: List[Score], cltype: int, index: int) -> List[List[int]]:
+        scorestruct: Dict[int, List[int]] = {}
+
+        for score in scores:
+            musicid = score.id
+            chart = score.chart
+
+            # Filter to only singles/doubles charts
+            if cltype == self.CLEAR_TYPE_SINGLE:
+                if chart not in [
+                    self.CHART_TYPE_B7,
+                    self.CHART_TYPE_N7,
+                    self.CHART_TYPE_H7,
+                    self.CHART_TYPE_A7,
+                    self.CHART_TYPE_L7,
+                ]:
+                    continue
+                chartindex = {
```

Comment:
My mistake originally on all of this software (IIDX was one of the earliest ones I did and so you can see some old paradigms in the code), but this should be in a lookup function similar to the game_to_db_rank/db_to_game_rank functions above. The other mapping in base should move here, and we need to update the games before HV to include their own copypasta constants with unit mappings.